test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -1244,6 +1360,53 @@ private static IEdmModel GetEdmModel()
             return edmModel;
         }
 
+        /// <summary>
+        /// This large EdmModel is used to test issues related to writing large metadata documents. 
+        /// For example, async writing of large metadata documents, writing large metadata documents multiple times, etc.
+        /// </summary>
+        /// <returns>Large EdmModel</returns>
+        private static IEdmModel GetLargeEdmModel()
+        {
+            EdmModel edmModel = new EdmModel();
+
+            EdmEntityContainer container = new EdmEntityContainer("NS", "Container");
+            edmModel.AddElement(container);
+
+            string longString = GenerateLongString();
+
+            // Add 1000 entity types
+            for (int i = 0; i < 1000; i++)
+            {
+                EdmEntityType entityType = new EdmEntityType("NS", $"Entity{i}");
+                var idProperty = new EdmStructuralProperty(entityType, $"Entity{i}Id", EdmCoreModel.Instance.GetInt32(false));
+                entityType.AddProperty(idProperty);
+                entityType.AddKeys(new IEdmStructuralProperty[] { idProperty });
+
+                // Add 300 properties to each entity type
+                for (int j = 0; j < 100; j++)
+                {
+                    entityType.AddProperty(new EdmStructuralProperty(entityType, $"Property{DateTime.Now.Ticks}{j}", EdmCoreModel.Instance.GetString(false)));

```

Comment:
Why use current time ticks as property names? I think using a non-deterministic property name could potentially make tests harder to debug.

Reply:
@habbes This is just for testing purposes. The same way you would want to generate random long string (like 10000 characters long) to test a scenario. 

But you are right, I will use something like `638642487961128210`

Reply:
@WanjohiSammy  does have to be a number? Why not just make it a long string? Making it a random number might be confusing for the reader.